sklearn/compose/_column_transformer.py

```diff
@@ -227,14 +243,19 @@ def _validate_transformers(self):
 
     def _validate_remainder(self, X):
         """Generate list of passthrough columns for 'remainder' case."""
-        if self.remainder not in ('drop', 'passthrough'):
+        is_transformer = ((hasattr(self.remainder, "fit")
+                           or hasattr(self.remainder, "fit_transform"))
+                          and hasattr(self.remainder, "transform"))
```

Comment:
There is a small possibility that an estimator will not have `transform` until after fitting...

Reply:
I used the negation of condition in `_validate_transformers`:

```python
if (not (hasattr(t, "fit") or hasattr(t, "fit_transform")) or not
        hasattr(t, "transform")):
    raise TypeError("All estimators should implement fit and "
                    "transform, or can be 'drop' or 'passthrough' "
                    "specifiers. '%s' (type %s) doesn't." %
                    (t, type(t)))
```

I am curious. Which estimator defines `transform` dynamically after calling `fit`?  🤔

Reply:
The documentation in `ColumnTransformer` forces transformers to have  `transform` defined:
```
        transformer : estimator or {'passthrough', 'drop'}
            Estimator must support `fit` and `transform`. Special-cased
            strings 'drop' and 'passthrough' are accepted as well, to
            indicate to drop the columns or to pass them through untransformed,
            respectively.
```
It should be okay to apply this restriction to `remainder` as well.